tests/app/parser/test_supplementary_data_parser.py

```diff
@@ -54,6 +54,34 @@ def test_invalid_supplementary_data_payload_raises_error():
     assert str(error.value) == "Invalid supplementary data"
 
 
+def test_invalid_supplementary_dataset_version_raises_error():
+    with pytest.raises(ValidationError) as error:
+        validate_supplementary_data_v1(
+            supplementary_data=SUPPLEMENTARY_DATA_PAYLOAD,
+            dataset_id="44f1b432-9421-49e5-bd26-e63e18a30b69",
+            identifier="12345678901",
+            survey_id="123",
+            sds_schema_version="v6",
+        )
+
+    assert (
+        str(error.value)
+        == "{'_schema': ['The Supplementary Dataset Schema Version does not match the version set in the Questionnaire Schema']}"
+    )
+
+
+def test_valid_supplementary_dataset_version():
```

Comment:
You can create some more readable code by just writing a test where valid dataset version does not raise an error by using some helper function like this one:
```python
from contextlib import contextmanager

@contextmanager
def not_raises(exception):
  try:
    yield
  except exception:
    raise pytest.fail("DID RAISE {0}".format(exception))
```

Reply:
...and then use it as `with not_raises(ValidationError):`

Reply:
Added this in 👍